drivers/interrupt_controller/Kconfig.it8xxx2

```diff
@@ -9,3 +9,11 @@ config ITE_IT8XXX2_INTC
 	 instance of the shared interrupt driver. To conserve RAM set
 	 this value to the lowest practical value.
 	 this software interrupt default set on by device tree.
+
+config ITE_IT8XXX2_WUC
+	def_bool $(dt_nodelabel_has_compat,wuc,it8xxx2-wuc)
+	depends on SOC_IT8XXX2
+	help
+	 This option enables the wakeup controller interface for IT8XXX2
+	 family.
+	 This is required for GPIO, KSCAN, etc., interrupt support.
```

Comment:
def_bool FOO

is the same as

bool
default FOO

so it surprises me that you get undefined reference issues when you don't do

default y

elsewhere.

In this build that causes undefined reference issues, are you expecting $(dt_nodelabel_has_compat,wuc,it8xxx2-wuc)
to be true?

Is $(dt_nodelabel_has_compat,wuc,it8xxx2-wuc) ever true?

perhaps $(dt_nodelabel_has_compat,wuc,it8xxx2-wuc) is accidentally always false, and then we fix it with Kconfig.

If $(dt_nodelabel_has_compat,wuc,it8xxx2-wuc) is always false, then there is no point in using DT.

Reply:
You could test if dt is accidentally always false

by doing

def_bool n

and see if anything changes. If it does, then I am wrong, and I apologize.

Reply:
I upload my test code in https://github.com/zephyrproject-rtos/zephyr/commit/2028c92cc7eebba08a27f2180718796b26d04756 
to test "def_bool n", and it shows "undefined reference wuc driver functions", too, when build.  

No problem! Thank you for your reviewing on this PR.

Reply:
Sorry, I did not make myself clear.

Replacing $(dt_nodelabel_has_compat,wuc,it8xxx2-wuc) with n

and retaining the

config ITE_IT8XXX2_WUC
	default y

without there being any functional changes will prove that $(dt_nodelabel_has_compat,wuc,it8xxx2-wuc) is always false, which is not intended behaviour.

In your test code you also removed the config y, which does not prove anything interesting.

Reply:
Right, so this proves that $(dt_nodelabel_has_compat,wuc,it8xxx2-wuc) is evaluating to false.

Probably there is something wrong with the DT, because it should evaluate to true, right?

I have asked in Discord and found that

config ITE_IT8XXX2_WUC
default y

should be removed.

and that def_bool $(dt_nodelabel_has_compat,wuc,it8xxx2-wuc) should be used.

But you need to find out why the DT is being evaluated to false when it is expected to be true.

Reply:
Done